flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/resource/ResourceManager.java

```diff
@@ -152,6 +156,26 @@ public Set<URL> getLocalJarResources() {
                 .collect(Collectors.toSet());
     }
 
+    /**
+     * Adds the local jar resources to the given {@link TableConfig}. It implicitly considers the
+     * {@link TableConfig#getRootConfiguration()} and stores the merged result into {@link
+     * TableConfig#getConfiguration()}.
+     */
+    public void addJarConfiguration(TableConfig tableConfig) {
+        final List<String> jars =
+                getLocalJarResources().stream().map(URL::toString).collect(Collectors.toList());
+        if (jars.isEmpty()) {
+            return;
+        }
+        final Set<String> jarFiles =
+                tableConfig
+                        .getOptional(PipelineOptions.JARS)
+                        .map(HashSet::new)
```

Comment:
A `LinkedHashSet` may be better to retain the order as configured by users.

Reply:
Good catch!